src/Raven.Server/Https/ExternalCertificateValidator.cs

```diff
@@ -0,0 +1,288 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Http;
+using Sparrow;
+using Sparrow.Logging;
+using Sparrow.Utils;
+
+namespace Raven.Server.Https
+{
+    public class ExternalCertificateValidator
+    {
+        private readonly RavenServer _server;
+        private readonly Logger _logger;
+
+        private ConcurrentDictionary<Key, Task<CachedValue>> _externalCertificateValidationCallbackCache;
+
+        public ExternalCertificateValidator(RavenServer server, Logger logger)
+        {
+            _server = server;
+            _logger = logger;
+        }
+
+        public void Initialize()
+        {
+            if (string.IsNullOrEmpty(_server.Configuration.Security.CertificateValidationExec))
+                return;
+
+            _externalCertificateValidationCallbackCache = new ConcurrentDictionary<Key, Task<CachedValue>>();
```

Comment:
We should validate here that the file exists probably?

Reply:
It's a little tricky because the file can be the exec (if it's a .exe for example) or it can be any of the arguments, for example, if we run a script via Powershell or bash.

Instead, if the command won't succeed it will throw and log the error.